inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -837,84 +834,92 @@ void ReorderConcatInputsPass::run() {
     }
     int numOfLinkLayers = 0;
 
-    for (auto & l : *pLayers) {
-        // 1st stage locate concat align filter
+    for (auto& l : *pLayers) {
+        // 1st stage locate concat
         LayerInfo info(l);
-        if (!info.isConcatAlignFilter()) {
+        if (!info.isConcat()) {
```

Comment:
the whole purpose of align filters reordering looks misunderstood - for concat without overlap reodering not needeed at all any execution order should work.

Reply:
please go through the code more carefully. The first step is to find concat layer.
In the second step checking all inputs to concat layer to find concat align ones.